File: 92 - Orientacao a objetos/java/comum/Pessoa.java
public class Pessoa{
	//construtor
	public Pessoa(){
		this.nome_temp = "tmp nome";
	}

	//atributos publicos
	public String nome;
	public String sobrenome;
	public String cpf;




	//atributos privados
	private String nome_temp;




	//metodos publicos
	public void euSou(){
		System.out.println("eu sou pessoa");
	}
	//agregando metodos
	public void euSou(String nome){
		System.out.println("eu sou " + nome);
	}

	public void metodoSobresquever(){
		System.out.println("Metodo original");
	}




	//metodos privados
	private String concatenaNomes(){
		return this.nome + " " + this.sobrenome;
	}




	//metodos protegidos
	protected String nomeCompletoComCPF(){
		return this.nome + " " + this.sobrenome + ", CPF: " + this.cpf;
	}




	//metodos staticos
	public static String oQueEuSou(){
		return "Eu sou uma classe que herda de Pessoa";
	}
}

File: 1l - Orientacao a objetos/ruby/pessoa.rb
class Pessoa
	def initialize
		@nome_temp = "tmp nome"
	end

	public
	#atributos publicos
	attr_accessor :nome, :sobrenome, :cpf


	#atributos privados
	private 
	attr_accessor :nome_temp




	#metodos publicos
	public 
	def eu_sou
		puts("eu sou pessoa")
	end
	def metodo_sobresquever
		puts("Metodo original")
	end




	#metodos privados
	private 
	def concatenaNomes
		return self.nome + " " + self.sobrenome
	end




	#metodos protegidos
	protected 
	def nomeCompletoComCPF
		return self.nome + " " + self.sobrenome + ", CPF: " + self.cpf
	end




	#metodos staticos
	def self.o_que_eu_sou()
		return "Eu sou uma classe que herda de Pessoa"
	end
end